.github/workflows/e2e-tests.yml

```diff
@@ -0,0 +1,49 @@
+name: Run e2e tests on Kedro-Viz
+
+on:
+  workflow_call:
+    inputs:
+      os:
+        type: string
+      python-version:
+        type: string
+jobs:
+    e2e_tests:
+        runs-on: ${{ inputs.os }}
+        if: inputs.os == 'ubuntu-latest' || ((github.ref == 'refs/heads/main' || github.ref == 'refs/heads/demo') && inputs.os == 'windows-latest')
+        steps:
+        - name: Checkout code
+          uses: actions/checkout@v4
+
+        - name: Set up Python ${{inputs.python-version}}
+          uses: actions/setup-python@v5
+          with:
+            python-version: ${{inputs.python-version}}          
+
+        - name: Cache python packages for Linux
+          if: inputs.os == 'ubuntu-latest'
+          uses: actions/cache@v4
+          with:
+            path: ~/.cache/pip
+            key: ${{inputs.os}}-python-${{inputs.python-version}}
+
+        - name: Cache python packages for Windows
+          if: inputs.os == 'windows-latest'
+          uses: actions/cache@v4
+          with:
+            path: ~\AppData\Local\pip\Cache
+            key: ${{inputs.os}}-python-${{inputs.python-version}}
+
+        - name: Setup Python environment
```

Comment:
Is this required?

Reply:
In CircleCI here https://github.com/jitu5/kedro-viz/blob/main/.circleci/continue_config.yml#L143

"Setup Python environment" is one of the step for e2e-tests.

Reply:
But the second step above already sets up python ^

Reply:
I realised it actually installs kedro and other requirements, maybe rename the action.